src/%IPM/Main.cls

```
@@ -2383,11 +2408,208 @@ ClassMethod UpdateLanguageExtensions(pVerbose As %Boolean = 0, pTestOnly As %Boo
+			}
+		}
+	}
+	Set $Namespace = baseNS
+	If globally {
+		Merge targetNamespaces = namespacesWithoutIPM
+	} ElseIf (namespaces '= "") {
+		Set specifiedNSs = $ListFromString(namespaces, ",")
+		Set ptr = 0
+   		While $ListNext(specifiedNSs, ptr, value) {
+			If '$DATA(namespacesWithoutIPM(value)) {
+				Write !, value_" is either not a valid namespace to install IPM, or already has IPM installed. Skipped."
+			} Else {
+				Set targetNamespaces(value) = 1
+			}
+		}
+	} Else {
+		// TODO: Implement this part
+		Write !, "Currently do not support interactively choose namespaces to install IPM. Use -namespaces or -globally modifier."
+		Return
+	}
+	// Now that we got all the namespaces that need to install IPM, do the actual installation
+	If 'useLocal {
+		Set server = ##class(%IPM.Repo.Remote.Definition).DeploymentServerOpen(1,,.sc)
+		$$$ThrowOnError(sc)
+		If $IsObject(server) {
+			Set packageService = server.GetPackageService()
+			Set ipmRef = ##class(%IPM.Storage.ModuleInfo).%New()
+			// TODO: Make this ipm eventually once the package name is changed
+			Set ipmRef.Name = "zpm"
+			Set ipmRef.VersionString = targetVersion
+			
+			// Retrieve module installer manifest
+			Set manifest = packageService.GetModuleInstallerManifest(ipmRef)
+			Set currentNS = ""
+			While 1 {
+				Set currentNS = $ORDER(targetNamespaces(currentNS))
+				If currentNS = "" {
+					Quit
+				}
+				New $Namespace
+				Set $Namespace = currentNS
+				Do $SYSTEM.OBJ.LoadStream(manifest, "ck")
+				Write !, "IPM enabled for namespace "_currentNS
+			}
+		} Else {
+			$$$ThrowOnError($$$ERROR($$$GeneralError,"Failed to get a valid registry server in order to download the IPM manifest."))
+		}
+	} Else {
+		Set targetIPMFileName = ipmLocalArray(targetVersion)
+		Set currentNS=$ORDER(targetNamespaces(""))
+		While (currentNS '="") {
+			Set $Namespace = currentNS
+			Do $SYSTEM.OBJ.Load(##class(%File).NormalizeFilename(targetIPMFileName, XMLDir), "ck")
```

Should accumulate status per namespace and only show IPM enabled message if was successful. Should abandon if IPM failed to install in any namespace instead of continuing.

---

Also, use the `-d` flag so that minimal output is displayed (only see errors and not list of classes loaded)